File: leet_383_ransom_note___2pass.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine)
    {
        vector<int>     mp(26, 0);

        for (char c : magazine)
            mp[c - 'a']++;
        for (char c : ransomNote)
        {
            mp[c - 'a']--;
            if (mp[c - 'a'] < 0)
                return false;
        }
        return true;
    }
};

File: leet_383_ransom_note___2pass.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mp = [0] * 26
        a = ord('a')
        for c in magazine:
            mp[ord(c) - a] += 1
        for c in ransomNote:
            i = ord(c) - a
            mp[i] -= 1
            if mp[i] < 0:
                return False
        return True
